lib/Service/CircleService.php

```diff
@@ -555,15 +555,20 @@ private function confirmSanitizedName(Circle $circle) {
 	 * @return string
 	 */
 	public function sanitizeName(string $name): string {
-		$acceptedChars = 'qwertyuiopasdfghjklzxcvbnm1234567890 ';
-		$sanitized = '';
-		for ($i = 0; $i < strlen($name); $i++) {
-			if (strpos($acceptedChars, strtolower($name[$i])) !== false) {
-				$sanitized .= $name[$i];
-			}
-		}
+		// replace '/' with '-' to prevent directory traversal
+		// replacing instead of stripping seems the better tradeoff here
+		$sanitized = str_replace('/', '-', $name);
+
+		// remove characters which are illegal on Windows (includes illegal characters on Unix/Linux)
+		// see also \OC\Files\Storage\Common::verifyPosixPath(...)
+		/** @noinspection CascadeStringReplacementInspection */
+		$sanitized = str_replace(['*', '|', '\\', ':', '"', '<', '>', '?'], '', $sanitized);
+
+		// remove leading spaces or dots to prevent hidden files
+		$sanitized = preg_replace('/^[\. ]+/mu', '', $sanitized);
```

Comment:
might be useless if you replace L571: `return trim($sanitized, '. ');`

Reply:
The diference would be that dots (`.`) would be removed from the end of the string as well. While I agree that they're not particularly useful, I still would stick to the minimal technically necessary changes.

I don't have hard feelings here though :wink:

Reply:
Changed to your suggested solution now :)